lightning/src/ln/channelmanager.rs

```diff
@@ -9011,15 +9015,31 @@ where
 				let peer_state = &mut *peer_state_lock;
 				let pending_msg_events = &mut peer_state.pending_msg_events;
 
-				peer_state.channel_by_id.iter_mut().filter_map(|(_, phase)|
-					if let ChannelPhase::Funded(chan) = phase { Some(chan) } else { None }
-				).for_each(|chan| {
-					let logger = WithChannelContext::from(&self.logger, &chan.context);
-					pending_msg_events.push(events::MessageSendEvent::SendChannelReestablish {
-						node_id: chan.context.get_counterparty_node_id(),
-						msg: chan.get_channel_reestablish(&&logger),
-					});
-				});
+				for (_, phase) in peer_state.channel_by_id.iter_mut() {
+					match phase {
+						ChannelPhase::Funded(chan) => {
+							let logger = WithChannelContext::from(&self.logger, &chan.context);
+							pending_msg_events.push(events::MessageSendEvent::SendChannelReestablish {
+								node_id: chan.context.get_counterparty_node_id(),
+								msg: chan.get_channel_reestablish(&&logger),
+							});
+						}
+
+						ChannelPhase::UnfundedOutboundV1(chan) => {
+							pending_msg_events.push(events::MessageSendEvent::SendOpenChannel {
+								node_id: chan.context.get_counterparty_node_id(),
+								msg: chan.get_open_channel(self.chain_hash),
+							});
+						}
+
+						ChannelPhase::UnfundedInboundV1(_) => {
+							// Since unfunded inbound channel maps are cleared upon disconnecting a peer, and they're not
+							// persisted (so won't be recovered after a crash), they shouldn't exist here and we would never
+							// need to worry about closing and removing them.
```

Comment:
This sentence is a bit lengthy, and the second portion isn't accurate as we aren't doing clean-up here.

Let's re-phrase and tighten this up a bit as:

Since unfunded inbound channel maps are cleared upon disconnecting a peer, they're not persisted so won't be recovered after a crash. Therefore, they shouldn't exist here.

Reply:
Addressed it in [f9fbdca](https://github.com/lightningdevkit/rust-lightning/commit/f9fbdcad5e018c8533fc3670bba3dd4b9fd281ce)
Thanks for the suggestion on rephrasing!